x-pack/plugins/apm/public/components/shared/TransactionActionMenu/TransactionActionMenu.tsx

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiButtonEmpty,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiLink,
+  EuiPopover
+} from '@elastic/eui';
+import React from 'react';
+import { oc } from 'ts-optchain';
+import { getUnconnectedKibanaHref } from 'x-pack/plugins/apm/public/utils/url';
+import { Transaction } from 'x-pack/plugins/apm/typings/es_schemas/Transaction';
+import { getDiscoverQuery } from '../DiscoverButtons/DiscoverTransactionButton';
+
+function getInfraMetricsQuery(transaction: Transaction) {
+  const plus5 = new Date(transaction['@timestamp']);
+  const minus5 = new Date(transaction['@timestamp']);
+
+  plus5.setMinutes(plus5.getMinutes() + 5);
+  minus5.setMinutes(minus5.getMinutes() - 5);
+
+  return {
+    from: minus5.getTime(),
+    to: plus5.getTime()
+  };
+}
+
+function ActionMenuButton({ onClick }: { onClick: () => void }) {
+  return (
+    <EuiButtonEmpty iconType="arrowDown" iconSide="right" onClick={onClick}>
+      Actions
+    </EuiButtonEmpty>
+  );
+}
+
+interface TransactionActionMenuProps {
+  readonly transaction: Transaction;
+  readonly location: Location;
+}
+
+interface TransactionActionMenuState {
+  readonly isOpen: boolean;
+}
+
+export class TransactionActionMenu extends React.Component<
+  TransactionActionMenuProps,
+  TransactionActionMenuState
```

Comment:
Can we just name these `Props` and `State` or are there other similar interfaces causing naming conflicts/confusion?